src/Pool.sol

```diff
@@ -224,7 +154,7 @@ abstract contract Pool {
         Receiver storage receiver = receivers[msg.sender];
         uint64 collectedCycle = receiver.nextCollectedCycle;
         if (collectedCycle == 0) return 0;
-        uint64 currFinishedCycle = currTimestamp() / cycleSecs;
+        uint64 currFinishedCycle = _currTimestamp() / cycleSecs;
```

Comment:
I think it might be nice to have individual functions for the calculate cycle operations based on timestamp operations. Instead of having one for the current timestamp.

Reply:
It's a good idea, but I'd rather port the tests before doing any refactoring, what do you think?